WebAPI/Controllers/ProductsController.cs

```diff
@@ -22,4 +22,43 @@ public async Task<ActionResult<List<ProductDetailDto>>> GetAll()
 
         return Ok(result);
     }
+
+    [HttpPost]
+    public async Task<ActionResult<Task>> Add(ProductDetailDto product)
+    {
+        var taskResult = _productService.AddAsync(product);
+
+        if (taskResult.IsCompletedSuccessfully)
+        {
+            return Ok(taskResult);
+        }
+
+        return BadRequest(taskResult);
+    }
+
+    [HttpPost]
+    public async Task<ActionResult<Task>> Update(ProductDetailDto product)
+    {
+        var taskResult = _productService.UpdateAsync(product);
+
+        if (taskResult.IsCompletedSuccessfully)
+        {
+            return Ok(taskResult);
+        }
+
+        return BadRequest(taskResult.Status);
+    }
+
+    [HttpDelete]
+    public async Task<ActionResult<Task>> Delete(ProductDetailDto product)
```

Comment:
For `DELETE` method I would definitely advise to use query-params for passing data since `DELETE` is not supporting request body (at least it is not defined yet by Web RFC). Also, you don't need to pass full model to remove entity, it is enough just id. Let's update your action and service to accept only id